st7735/st7735.go

```diff
@@ -0,0 +1,442 @@
+		d.Data(0x00)
+		d.Data(0x02)
+		d.Data(0x00)
+		d.Data(0x7F + 0x02)
+		d.Command(RASET)
+		d.Data(0x00)
+		d.Data(0x01)
+		d.Data(0x00)
+		d.Data(0x9F + 0x01)
+	} else if d.model == MINI80x160 {
+		d.isBGR = true
+		d.InvertColors(true)
+		d.Command(CASET)
+		d.Data(0x00)
+		d.Data(0x00)
+		d.Data(0x00)
+		d.Data(0x7F)
+		d.Command(RASET)
+		d.Data(0x00)
+		d.Data(0x00)
+		d.Data(0x00)
+		d.Data(0x9F)
+	}
+
+	// common color adjustment
+	d.Command(GMCTRP1)
+	d.Data(0x02)
+	d.Data(0x1C)
+	d.Data(0x07)
+	d.Data(0x12)
+	d.Data(0x37)
+	d.Data(0x32)
+	d.Data(0x29)
+	d.Data(0x2D)
+	d.Data(0x29)
+	d.Data(0x25)
+	d.Data(0x2B)
+	d.Data(0x39)
+	d.Data(0x00)
+	d.Data(0x01)
+	d.Data(0x03)
+	d.Data(0x10)
+	d.Command(GMCTRN1)
+	d.Data(0x03)
+	d.Data(0x1D)
+	d.Data(0x07)
+	d.Data(0x06)
+	d.Data(0x2E)
+	d.Data(0x2C)
+	d.Data(0x29)
+	d.Data(0x2D)
+	d.Data(0x2E)
+	d.Data(0x2E)
+	d.Data(0x37)
+	d.Data(0x3F)
+	d.Data(0x00)
+	d.Data(0x00)
+	d.Data(0x02)
+	d.Data(0x10)
+
+	d.Command(NORON)
+	time.Sleep(10 * time.Millisecond)
+	d.Command(DISPON)
+	time.Sleep(500 * time.Millisecond)
+
+	if cfg.Model == MINI80x160 {
+		d.Command(MADCTL)
+		d.Data(0xC0)
+	}
+
+	d.SetRotation(d.rotation)
+
+	d.blPin.High()
+}
+
+// Display does nothing, there's no buffer as it might be too big for some boards
+func (d *Device) Display() error {
+	return nil
+}
+
+// SetPixel sets a pixel in the screen
+func (d *Device) SetPixel(x int16, y int16, c color.RGBA) {
+	if x < 0 || y < 0 ||
+		(((d.rotation == NO_ROTATION || d.rotation == ROTATION_180) && (x >= d.width || y >= d.height)) ||
+			((d.rotation == ROTATION_90 || d.rotation == ROTATION_270) && (x >= d.height || y >= d.width))) {
+		return
+	}
+	d.FillRectangle(x, y, 1, 1, c)
+}
+
+// setWindow prepares the screen to be modified at a given rectangle
+func (d *Device) setWindow(x, y, w, h int16) {
+	if d.rotation == NO_ROTATION || d.rotation == ROTATION_180 {
+		x += d.columnOffset
+		y += d.rowOffset
+	} else {
+		x += d.rowOffset
+		y += d.columnOffset
+	}
+	d.Tx([]uint8{CASET}, true)
+	d.Tx([]uint8{uint8(x << 8), uint8(x), uint8((x + w - 1) << 8), uint8(x + w - 1)}, false)
+	d.Tx([]uint8{RASET}, true)
+	d.Tx([]uint8{uint8(y << 8), uint8(y), uint8((y + h - 1) << 8), uint8(y + h - 1)}, false)
+	d.Command(RAMWR)
+}
+
+// FillRectangle fills a rectangle at a given coordinates with a color
+func (d *Device) FillRectangle(x, y, width, height int16, c color.RGBA) error {
+	if x < 0 || y < 0 || width <= 0 || height <= 0 ||
+		(((d.rotation == NO_ROTATION || d.rotation == ROTATION_180) &&
+			(x >= d.width || y >= d.height || (x+width) > d.width || (y+height) > d.height)) ||
+			((d.rotation == ROTATION_90 || d.rotation == ROTATION_270) &&
+				(x >= d.height || y >= d.width || (x+width) > d.height || (y+height) > d.width))) {
+		return errors.New("rectangle coordinates outside display area")
+	}
+	d.setWindow(x, y, width, height)
+	c565 := RGBATo565(c)
+	c1 := uint8(c565 >> 8)
+	c2 := uint8(c565)
+
+	data := make([]uint8, d.batchLength*2)
```

Comment:
Because `d.batchLength` is not a constant I think it would be better to allocate this slice in the `Configure` call to avoid unnecessary heap allocations.

The same goes for the `make` call in `FillRectangleWithBuffer` below.